### ISSUE TITLE
Do not write image list to args.json files

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-server#1237]
Related to #474.

TLDR: let's remove the `args["metadata"]["image"]` key-value pair before writing args.json files to disk.

---

Here is a real-life example (job ID 1029):

- The workflow has nine tasks (from 0 to 8).
- The dataset has 357 wells and 26 cycles, corresponding to 9282 images (these values are read for the last task).
- The job folder uses approximately 25 GB.
- The largest files in the folder are the `args.json` files, which are 38557 (not so far from 4x9282=37128).
- Each one of these `args.json` file has a size of about 600K, and their total occupied size is then around 23 G. This the largest part of the whole job folder, then we'll ignore the rest.
- The different contribution to these 600K are listed below, using the approximate size estimates from https://pympler.readthedocs.io:

See script:
```python
import json
from pympler.asizeof import asizeof

FILENAME = "something.args.json"
with open(FILENAME, "r") as f:
    args = json.load(f)
args_size = asizeof(args)
print(f"args: {args_size}")
print("-" * 80)
for key, value in args.items():
    size_kB = asizeof(value) / 1000
    print(f"metadata    {key:22s}: size ~ {size_kB:07.2f} KB")
print("-" * 80)
for key, value in args["metadata"].items():
    try:
        len_value = len(value)
    except TypeError:
        len_value = 0
    size_kB = asizeof(value) / 1000
    print(f"metadata    {key:22s}: size ~ {size_kB:07.2f} KB (len={len_value})")
print("-" * 80)
```
and output:
```
args: 1092032
--------------------------------------------------------------------------------
metadata    input_paths           : size ~ 0000.21 KB
metadata    output_path           : size ~ 0000.12 KB
metadata    metadata              : size ~ 1089.17 KB
metadata    contrast_threshold    : size ~ 0000.03 KB
metadata    level                 : size ~ 0000.02 KB
metadata    overwrite             : size ~ 0000.03 KB
metadata    label_image_name      : size ~ 0000.06 KB
metadata    channel               : size ~ 0000.35 KB
metadata    label_image_cycle     : size ~ 0000.02 KB
metadata    intensity_image_cycle : size ~ 0000.03 KB
metadata    min_threshold         : size ~ 0000.03 KB
metadata    max_threshold         : size ~ 0000.03 KB
metadata    gaussian_blur         : size ~ 0000.03 KB
metadata    fill_holes_area       : size ~ 0000.03 KB
metadata    output_label_cycle    : size ~ 0000.03 KB
metadata    output_label_name     : size ~ 0000.06 KB
metadata    component             : size ~ 0000.10 KB
--------------------------------------------------------------------------------
metadata    plate                 : size ~ 0000.18 KB (len=1)
metadata    well                  : size ~ 0037.53 KB (len=357)
metadata    image                 : size ~ 1041.00 KB (len=9282)
metadata    num_levels            : size ~ 0000.03 KB (len=0)
metadata    coarsening_xy         : size ~ 0000.03 KB (len=0)
metadata    original_paths        : size ~ 0009.46 KB (len=26)
metadata    image_extension       : size ~ 0000.06 KB (len=3)
metadata    image_glob_patterns   : size ~ 0000.02 KB (len=0)
--------------------------------------------------------------------------------
```

It's clear that the `args["metadata"][["image"]` value (a list of 9282 strings) is the heaviest contributor to the file size.

After removing the image list, the size of one specific args.json file goes from 613K to 22K. This leads to a drastic size decrease of the args.json contribution, from about 23G to <1G.

**Comment by tcompa** [comment]
For the record, this would break the copy-ome-zarr task in fractal-tasks-core, see https://github.com/fractal-analytics-platform/fractal-tasks-core/blob/main/fractal_tasks_core/tasks/copy_ome_zarr.py#L210-L215:
```python
    for key in ["plate", "well", "image"]:
        meta_update[key] = [
            component.replace(".zarr", f"_{suffix}.zarr")
            for component in metadata[key]
        ]
```

Which means that we still need to include the image list, when working with this specific task. I guess the easiest is to have a boolean flag (e.g. `include_image_list`) internally that is always `False` and only set to `True` when a specific condition is met. The condition would be like: task name is Copy OME Zarr _and_ task source include `fractal_tasks_core`. We can include more cases if we find them.
